### PR TITLE
Feature/player controls

### PR DESCRIPTION
**Pull request by hugurp**
- move controls down
- prepare for provider
- add mute/unMute

note: there are no event thrown by the Youtube iframe api related to `volume change` or `muted` state. Therefore it is not possible to to react in the `<radio4000-player>` UI changes occurring when a user interacts with sound directly on the Youtube iframe.

### CODE REVIEW

**Comment by hugurp** [comment]
Here they have some event listeners for `volumeChanged`.

https://github.com/videojs/videojs-youtube/blob/master/src/Youtube.js#L235

Just tried with the youtube-player npm module, does not send them.

Maybe the yt iframe script directly will

**Comment by hugurp** [comment]
volume working here http://jsbin.com/kukicituca/edit?html,console,output

File: src/PlayerControls.vue [hunk 1]
```diff
@@ -1,27 +1,46 @@
...
-		<!-- <button @click="$emit('mute')">Mute</button> -->
-		<!-- <button @click="$emit('unMute')">Unmute</button> -->
+
+		<!-- https://en.wikipedia.org/wiki/Geometric_Shapes -->
+		<div class="PlayerControl-group">
+			<button v-if="!isMute"
+							@click="$emit('mute')"
+							v-bind:class="{ 'Btn--isNotFullVolume' : isNotFullVolume }"
+							class="Btn">
+				<span>■</span>
```
Comment (oskarrough) [inline src/PlayerControls.vue]: how about a volume icon?

File: src/PlayerControls.vue [hunk 2]
```diff
@@ -1,27 +1,46 @@
...
-			mute() { this.$emit('setVolume', 0) },
-			unMute() { this.player.unMute() },
-			changeVolume(vol) {
-				this.$emit('volumeChanged', vol)
-			}
-			*/
-		}
+		props: ['volume',
+						'isPlaying',
+						'isMute',
```
Comment (oskarrough) [inline src/PlayerControls.vue]: Can we keep the `isMuted` naming everywhere? Then it's the same as the YT API.

File: src/Radio4000Player.vue [hunk 3]
```diff
@@ -69,63 +77,67 @@
...
 			}
 		},
-		methods: {
-			selectTrack(track) {
-				this.autoplay = true
-				this.cueTrack(track)
+		watch: {
+			track: function(track) {
+				this.playTrack(track);
+				return track;
```
Comment (oskarrough) [inline src/Radio4000Player.vue]: watch methods should not return anything

File: src/Radio4000Player.vue [hunk 4]
```diff
@@ -69,63 +77,67 @@
...
-				this.playing = false
-				this.next()
+			getNextTrack() {
+				const pool = this.tracksPool
+				const index = pool.indexOf(this.currentTrack)
+				return pool[index + 1]
 			},
 			play() {
-				this.playing = true
+				this.isPlaying = true;
```
Comment (oskarrough) [inline src/Radio4000Player.vue]: no semi

File: src/Radio4000Player.vue [hunk 5]
```diff
@@ -69,63 +77,67 @@
...
-				this.next()
+			newTracksPool(tracks) {
+				this.tracksPool = tracks;
 			},
-			onPlayerPlaying(event) {
-				this.playing = true
+			playNextTrack() {
+				const track = this.getNextTrack()
+				console.log('playNextTrack:track', track)
+				if (!track) {
```
Comment (oskarrough) [inline src/Radio4000Player.vue]: missing `loop` integration?

File: src/Radio4000Player.vue [hunk 6]
```diff
@@ -69,63 +77,67 @@
...
-				if (!track && this.loop) {
-					track = playlist[0]
-				}
-				if (!track) {
-					return
-				}
-				this.autoplay = true
-				this.cueTrack(track)
+			mute() {
+				this.isMute = true;
```
Comment (oskarrough) [inline src/Radio4000Player.vue]: `isMuted`

File: src/YoutubePlayer.vue [hunk 7]
```diff
@@ -6,93 +6,148 @@
...
-				fs: 0,
-				modestbranding: 1,
-				origin: window.location.origin,
-				playsinline: 1,
-				rel: 0,
-				showinfo: 0
+	export default {
+		name: 'youtube-player',
+		props: [
+			'videoId',
```
Comment (oskarrough) [inline src/YoutubePlayer.vue]: shouldn't `videoId` be removed if we only accept `track` now?

File: src/YoutubePlayer.vue [hunk 8]
```diff
@@ -6,93 +6,148 @@
...
+			'volume',
+			'autoplay',
+			'isPlaying',
+			'isMute',
```
Comment (oskarrough) [inline src/YoutubePlayer.vue]: isMuted

File: src/Radio4000Player.vue [hunk 9]
```diff
@@ -69,63 +77,67 @@
...
-				this.track = track
+			tracks: function(tracks) {
+				this.newTracksPool(tracks);
+				return tracks;
 			},
-			// runs once on load when yt-iframe is ready
-			onPlayerReady(player) {
-				this.playerReady = true
+			volume: function(volume) {
+				if(volume <= 0) {
```
Comment (oskarrough) [inline src/Radio4000Player.vue]: shouldn't this be

```js
if (volume > 0) {
  mute()
} else {
  unMute()
}
```

File: src/Radio4000Player.vue (hunk 4 above)
Comment (hugurp) [inline src/Radio4000Player.vue]: …

File: src/Radio4000Player.vue (hunk 5 above)
Comment (hugurp) [inline src/Radio4000Player.vue]: loop?

File: src/YoutubePlayer.vue [hunk 10]
```diff
@@ -128,7 +130,7 @@
 
 			// select track to play
 			setTrackOnProvider(track) {
-				const ytid = track.ytid;
+				const ytid = trackId;
 				this.player.loadVideoById({
 					'videoId': ytid
 				}).then(this.playProvider())
```
Comment (oskarrough) [inline src/YoutubePlayer.vue]: should just be `}).then(this.playProvider)`

File: src/Radio4000Player.vue (hunk 5 above)
Comment (oskarrough) [inline src/Radio4000Player.vue]: loop the playlist so it starts over when at last track

**Pull request merged**